flatland/core/graph/graph_simplification.py

```diff
@@ -0,0 +1,83 @@
+from typing import List, Tuple
+
+import networkx as nx
+from attr import attrs, attrib
+
+from flatland.core.graph.grid_to_graph import GraphTransitionMap
+
+GridNode = Tuple[int, int, int]  # row, column, heading (at cell entry)
+
+
+@attrs
+class DecisionPointGraphEdgeData:
+    """
+    The edge data of a decision point overlay graph.
+
+    Attributes
+    ----------
+    path: List[GridNode]
+        The list of collapsed cells, starting on a facing switch (e.g. where a decision has been taken upon entering).
+    len: int
+        Number of collapsed cells.
+
+    """
+    path = attrib(type=List[GridNode])
+    len = attrib(type=int)
+
+
+# TODO naming: decision point graph, simplified graph, collapsed graph, ....?
+class DecisionPointGraph:
```

Comment:
@aiAdrian is this an appropriate naming? decision point graph, simplified graph, collapsed graph, ....?